composer/datasets/lm_datasets.py

```diff
@@ -153,17 +179,19 @@ def initialize_object(self, batch_size: int, dataloader_hparams: DataLoaderHpara
         dataset = lm_datasets
 
         # for some tokenizers, e.g. GPT-2, they don't have padding tokens. Hence, we cannot use the LM collator.
-        if self.tokenizer.pad_token_id is None:
+        if tokenizer.pad_token_id is None:
             data_collator = transformers.default_data_collator
         else:
-            data_collator = transformers.DataCollatorForLanguageModeling(tokenizer=self.tokenizer,
+            data_collator = transformers.DataCollatorForLanguageModeling(tokenizer=tokenizer,
                                                                          mlm=self.use_masked_lm,
                                                                          mlm_probability=self.mlm_probability)
 
-        sampler = dist.get_sampler(dataset, drop_last=self.drop_last, shuffle=self.shuffle)
+        sampler = dist.get_sampler(cast(torch.utils.data.Dataset, dataset),
```

Comment:
can you add a comment mentioning that a HF dataset does not subclass a torch dataset?

Reply:
Done!